packages/sandbox/src/sandbox_executor.ts

```diff
@@ -64,7 +65,19 @@ export class AmplifySandboxExecutor {
   private getSecretLastUpdated = async (
     backendId: BackendIdentifier
   ): Promise<Date | undefined> => {
-    const secrets = await this.secretClient.listSecrets(backendId);
+    let secrets = undefined;
+    try {
+      secrets = await this.secretClient.listSecrets(backendId);
+    } catch (err) {
+      throw new AmplifyUserError(
```

Comment:
This should most likely be a fault

Reply:
The only fault I saw here was expired credentials. Let me change this to look inside the `SecretError`.

Reply:
At the very least, a 5xx from the service is always possible. Things like throttling should also be faults, not errors

Reply:
yup, I changed it to fault except for one case where the underlying exception is `ExpiredTokenException`